vmsyscall/src/lib.rs

```diff
@@ -1,94 +1,98 @@
+pub const WRITE_BUF_LEN: usize = 4000;
+
+/// The syscalls for the Hypervisor <-> VM syscall proxy
+#[allow(clippy::large_enum_variant, missing_docs)]
 pub enum VmSyscall {
-    /// int madvise(void *addr, size_t length, int advice);
+    Read {
+        fd: u32,
+        count: usize,
+    },
+    Write {
+        fd: u32,
+        count: usize,
+        data: [u8; WRITE_BUF_LEN],
+    },
     Madvise {
-        /// see madvise(2)
         addr: usize,
-        /// see madvise(2)
         length: usize,
-        /// see madvise(2)
         advice: i32,
     },
-    /// void *mmap(void *addr, size_t length, int prot, int flags, …);
     Mmap {
-        /// see mmap(2)
         addr: usize,
-        /// see mmap(2)
         length: usize,
-        /// see mmap(2)
         prot: i32,
-        /// see mmap(2)
         flags: i32,
     },
-    /// void *mremap(void *old_address, size_t old_size, size_t new_size, int flags, ... /* void *new_address */);
     Mremap {
-        /// see mremap(2)
         old_address: usize,
-        /// see mremap(2)
         old_size: usize,
-        /// see mremap(2)
         new_size: usize,
-        /// see mremap(2)
         flags: i32,
     },
-    /// int munmap(void *addr, size_t length);
     Munmap {
-        /// see munmap(2)
         addr: usize,
-        /// see munmap(2)
         length: usize,
     },
-    /// int mprotect(void *addr, size_t len, int prot);
     Mprotect {
-        /// see mprotect(2)
         addr: usize,
-        /// see mprotect(2)
         length: usize,
-        /// see mprotect(2)
         prot: i32,
     },
     // Todo: extend with needed hypervisor proxy syscalls
 }
 
 /// The return value of the syscalls to be serialized/deserialized via serde
 /// for the Hypervisor <-> VM syscall proxy
-#[derive(Serialize, Deserialize, Debug)]
+#[allow(clippy::large_enum_variant, missing_docs)]
 pub enum VmSyscallRet {
-    /// int madvise(void *addr, size_t length, int advice);
+    Read(Result<(i32, [u8; WRITE_BUF_LEN]), Error>),
+    Write(Result<i32, Error>),
     Madvise(Result<i32, Error>),
-    /// void *mmap(void *addr, size_t length, int prot, int flags, …);
     Mmap(Result<usize, Error>),
-    /// void *mremap(void *old_address, size_t old_size, size_t new_size, int flags, ... /* void *new_address */);
     Mremap(Result<usize, Error>),
-    /// int munmap(void *addr, size_t length);
     Munmap(Result<i32, Error>),
-    /// int mprotect(void *addr, size_t len, int prot);
     Mprotect(Result<i32, Error>),
 }
 
-/// The error codes of the syscalls to be serialized/deserialized via serde
-/// for the Hypervisor <-> VM syscall proxy
-#[derive(Debug, Serialize, Deserialize, PartialEq)]
+/// The error codes of the syscalls for the Hypervisor <-> VM syscall proxy
+#[allow(missing_docs)]
+#[derive(Debug, PartialEq)]
 pub enum Error {
-    /// standard error
     Errno(i64),
-    /// serialize error
     SerializeError,
-    /// deserialize error
     DeSerializeError,
 }
```

Comment:
I don't think Ser/DeSer errors are used anymore. Which also raises the question about whether this enum is needed.

Reply:
correct